src/main/java/org/spongepowered/api/data/DataStore.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * This file is part of SpongeAPI, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered <https://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.api.data;
+
+import com.google.common.reflect.TypeToken;
+import org.spongepowered.api.data.value.Value;
+
+public interface DataStore {
+
+    TypeToken<?> getSupportedToken();
+
+    DataView serialize(Iterable<Value<?>> values, DataView view);
```

Comment:
Why does this serialize/deserialize iterables instead of simple values?

Reply:
That's what it's getting, an `Iterable` of `Value`s...

Reply:
I mean `DataView serialize(Value<?> value, DataView view);` instead of all values saved at once

Reply:
What's the point of getting only one `Value` when there's possibly multiple values being stored?

Reply:
Ah. Will see when you finish the basic jd (in which you will explain the saving model, etc.)

So the `Value` here actually contains both the actual data and the key, right? If that's the case, all goes very well.

Reply:
What's been added is also serializing by `DataManipulator` so it's easier to "get" only the usable keys that are wanted.